src/python/pants/backend/shell/shell_command.py

```diff
@@ -162,6 +164,9 @@ async def prepare_shell_command_process(
                     rationale=f"execute `{shell_command.alias}` {shell_command.address}",
                 )
 
+    extra_env = await Get(Environment, EnvironmentRequest(extra_env_vars))
+    command_env.update(extra_env)
```

Comment:
Tried setting `command_env = await Get(Environment, EnvironmentRequest(extra_env_vars))` from the start, but `Environment` is frozen so it doesn't allow the follow-up additions needed to set the tools path